### ISSUE TITLE
zipfile.Path.open() signature not correct

### ISSUE THREAD

**Issue by bomzheg** [python/typeshed#9966]
MRE:

```python
from typing import BinaryIO
from zipfile import Path as ZipPath


def load_results(zip_file: BinaryIO):
    zip_path = ZipPath(zip_file)
    for unpacked_file in zip_path.iterdir():
        if not unpacked_file.is_file():
            continue
        if unpacked_file.name == "results.json":
            with unpacked_file.open("r", encoding="utf8") as results_file:  # line 11
                print(type(results_file.read()))  # line 12
                continue
        if unpacked_file.name == "image.jpg":
            with unpacked_file.open("rb") as image:  # line 15
                print(type(image.read()))


if __name__ == '__main__':
    with open("example.zip", "rb") as f:
        load_results(f)
```
To run the script, please add an `example.zip` file with `results.json` (with any text in UTF-8, including Cyrillic symbols) and an `image.jpg` file to your working directory.

In PyCharm (which uses typeshed), I am receiving warnings about the following lines of code:

* Line 11: `"Unexpected argument"` regarding the encoding keyword argument.
* Line 15: `"Expected type 'Literal["r", "w"]', got 'Literal["rb"]' instead"` regarding the type of the file mode.

In both cases, the code appears to work correctly:
* Without the encoding keyword argument, I encounter a UnicodeDecodeError.
* The print statements show different types (`<class 'str'>` for line 12 and `<class 'bytes'>` for line 15) as expected when using `"r"` or `"rb"` file mode.

**Comment by spatinom** [comment]
hello! my partner and I are students trying to work on some open issues for a course we're in. We know typeshed doesn't really do assignments but we'd appreciate it if you'd give us the opportunity to create a PR for this!

**Comment by JelleZijlstra** [comment]
Go ahead, please submit a PR!

**Comment by spatinom** [comment]
@bomzheg what version of python were you using?

**Comment by bomzheg** [comment]
right now - python 3.11.0

but for test reason I switched for 3.10.7 and still have warnings.

**Comment by grigoriev-semyon** [comment]
may be fixed in #10033

**Comment by AlexWaygood** [comment]
The errors no longer reproduce with typeshed `main` -- probably fixed in https://github.com/python/typeshed/pull/10033 and/or #10861
